### ISSUE TITLE
The getter 'connectionStatus' was called on null

### ISSUE THREAD

**Issue by milosjovac** [shamblett/mqtt_client#87]
![image](https://user-images.githubusercontent.com/2808506/55589624-fa3f2600-5730-11e9-881e-88b27f14a307.png)

![image](https://user-images.githubusercontent.com/2808506/55589594-e398cf00-5730-11e9-9d06-27da0194ee9d.png)

Flutter

**Comment by shamblett** [comment]
Yep, I'll put a fix in

**Comment by noordawod** [comment]
pr #89 is meant to fix this, and few other NPEs. Awaiting the maintainer for merging.

**Comment by yolkhovyy** [comment]
I think I observe this also here (when connection gets lost, I guess):
```
  int publishMessage(
      String topic, MqttQos qualityOfService, typed.Uint8Buffer data,
      {bool retain = false}) {
    if (_connectionHandler.connectionStatus.state !=
        MqttConnectionState.connected) {
      throw ConnectionException(_connectionHandler.connectionStatus.state);
    }
    try {
      final PublicationTopic pubTopic = PublicationTopic(topic);
      return _publishingManager.publish(
          pubTopic, qualityOfService, data, retain);
    } on Exception catch (e) {
      throw InvalidTopicException(e.toString(), topic);
    }
  }
```

**Comment by shamblett** [comment]
Client updated as above, re-published at 5.5.3
